test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +785,96 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
 
         [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateIStringLocalizerProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            // Act & Assert
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name fr-FR", kvp.Key.Name));
+            }
+
+            using (new CultureReplacer("en-US", "en-US"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name en-US", kvp.Key.Name));
+            }
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_NameWithOnlyDisplayAttribute()
+        {
+            // Arrange & Act
+            var enumNameAndGroup = GetLocalizedEnumGrouopedDisplayNamesAndValues();
+
+            var enumOne = enumNameAndGroup.First(e => e.Value == "1");
```

Comment:
In general, avoid `Linq` in tests because the methods `throw` unreadable `Exceptions` and xUnit.net gives you everything you need. In this case,
``` c#
var enumOne = Assert.Single(enumNameAndGroup, e => e.Value == "1"); // guard
```